controller/admin_transactions_controller.php

```diff
@@ -296,11 +307,88 @@ private function do_action($action, $args)
 				}
 				// Clear $action status
 				$action = '';
+			break;
+			case 'add':
+				$errors = array();
+
+				$transaction_data = array(
+					'MT_ANONYMOUS'          => $this->request->is_set('u'),
+					'MT_USERNAME'           => $this->request->variable('username', '', true),
+					'MT_FIRST_NAME'         => $this->request->variable('first_name', '', true),
+					'MT_LAST_NAME'          => $this->request->variable('last_name', '', true),
+					'MT_PAYER_EMAIL'        => $this->request->variable('payer_email', '', true),
+					'MT_RESIDENCE_COUNTRY'  => $this->request->variable('residence_country', ''),
+					'MT_MC_GROSS'           => $this->request->variable('mc_gross', (float) 0),
+					'MT_MC_CURRENCY'        => $this->request->variable('mc_currency', ''),
+					'MT_MC_FEE'             => $this->request->variable('mc_fee', (float) 0),
+					'MT_PAYMENT_DATE_YEAR'  => $this->request->variable('payment_date_year', (int) $this->user->format_date(time(), 'Y')),
+					'MT_PAYMENT_DATE_MONTH' => $this->request->variable('payment_date_month', (int) $this->user->format_date(time(), 'n')),
+					'MT_PAYMENT_DATE_DAY'   => $this->request->variable('payment_date_day', (int) $this->user->format_date(time(), 'j')),
+					'MT_PAYMENT_TIME'       => $this->request->variable('payment_time', $this->user->format_date(time(), 'H:i:s')),
+					'MT_MEMO'               => $this->request->variable('memo', '', true),
+				);
+
+				if ($this->request->is_set_post('submit'))
+				{
+					try
+					{
+						$data_ary = $this->build_data_ary($transaction_data);
+						$this->ppde_actions->log_to_db($data_ary);
+
```

Comment:
You don't add more actions? such as autogroup, update stats, update raised amount?
All these features are available via the class `core_actions`

Reply:
I added these 4 actions now, are there any other actions that could be useful?
```
$this->ppde_actions->update_donor_stats();
$this->ppde_actions->update_overview_stats();
$this->ppde_actions->update_raised_amount();
$this->ppde_actions->donors_group_user_add();
```

Reply:
Can I call `$this->ppde_actions->donors_group_user_add();` if the user already is in the donor group without causing any issues or unnecessary log entries?

Reply:
Yes you can call `$this->ppde_actions->donors_group_user_add();`.
The phpBB function `group_user_add()` remove the user before adding it in the group

Reply:
You can also add the donor notification (not the admin notification).
The following code must be run when the MT is added
```
$this->ppde_actions->update_overview_stats();
$this->ppde_actions->update_raised_amount();
```

The following only if the donor is a member.
```
$this->ppde_actions->update_donor_stats();
$this->ppde_actions->donors_group_user_add();
$this->ppde_actions->notification->notify_donor_donation_received();
```